### DISCUSSION TITLE
'ExactMatchLevel' object has no attribute 'get_comparison'

### DISCUSSION THREAD

**Discussion by tombetinashe** [moj-analytical-services/splink#2349] [Q&A]
I am currently receiving this error in splink 4 "ExactMatchLevel' object has no attribute 'get_comparison".Your assistance will be highly appreciated.

extended_model_settings = {
    "unique_id_column_name": "record_id",
    "link_type": "link_only",
    "blocking_rules_to_generate_predictions": blocking_rules,
    "comparisons": [
        {
            "output_column_name": "Full name",
            "comparison_levels": [
                {
                    "sql_condition": "(firstname_l IS NULL OR firstname_r IS NULL) and (surname_l IS NULL OR surname_r IS NULL)",
                    "label_for_charts": "Null",
                    "is_null_level": True,
                },
                # full name match
                cll.ExactMatchLevel("full_name", term_frequency_adjustments=True),
                # typos - keep levels across full name rather than scoring separately
                cll.JaroWinklerLevel("full_name", 0.9),
                cll.JaroWinklerLevel("full_name", 0.8),
                # name switched
                cll.ColumnsReversedLevel("firstname", "surname"),
                # name switched + typo
                
                # single name match
                cll.ExactMatchLevel("firstname", term_frequency_adjustments=True),
                cll.ExactMatchLevel("surname", term_frequency_adjustments=True),
                # single name cross-match
                {
                    "sql_condition": "firstname_l = surname_r OR surname_l = firstname_r",
                    "label_for_charts": "single name cross-matches",
                },  # single name typos
                cll.JaroWinklerLevel("firstname", 0.9),
                cll.JaroWinklerLevel("surname", 0.9),
                # the rest
                cll.ElseLevel(),
            ],
        },
        cl.DateOfBirthComparison(
            "dob",
            input_is_string=True,),
        cll.ExactMatchLevel("gender"),
        
        
      
    ],
    "retain_intermediate_calculation_columns": True,
}

linker_detailed=Linker(dfs, extended_model_settings, db_api=DuckDBAPI())

**Comment by ADBond** [comment]
It looks like at the end of your settings you have accidentally put a comparison level instead of a comparison, you should have `cl.ExactMatch("gender")`.
